src/Elliptic/DiscontinuousGalerkin/ImposeBoundaryConditions.hpp

```diff
@@ -0,0 +1,180 @@
+template <typename Tag>
+struct Magnitude;
+}  // namespace Tags
+// IWYU pragma: no_forward_declare db::DataBox
+/// \endcond
+
+namespace Elliptic {
+namespace dg {
+namespace Actions {
+
+/*!
+ * \brief Packages data on external boundaries so that they represent
+ * homogeneous (zero) Dirichlet boundary conditions.
+ *
+ * This action imposes homogeneous boundary conditions on all fields in
+ * `system::impose_boundary_conditions_on_fields`. The fields are wrapped in
+ * `LinearSolver::Tags::Operand`. The result should be a subset of the
+ * `system::variables`. Because we are working with the linear solver operand,
+ * we cannot impose non-zero boundary conditions here. Instead, non-zero
+ * boundary conditions are handled as contributions to the linear solver source
+ * during initialization.
+ *
+ * \warning This actions works only for scalar fields right now. It should be
+ * considered a temporary solution and will have to be reworked for more
+ * involved boundary conditions.
+ *
+ * With:
+ * - `interior<Tag> =
+ *   Tags::Interface<Tags::BoundaryDirectionsInterior<volume_dim>, Tag>`
+ * - `exterior<Tag> =
+ *   Tags::Interface<Tags::BoundaryDirectionsExterior<volume_dim>, Tag>`
+ *
+ * Uses:
+ * - Metavariables:
+ *   - `normal_dot_numerical_flux`
+ *   - `temporal_id`
+ * - System:
+ *   - `volume_dim`
+ *   - `variables_tag`
+ *   - `impose_boundary_conditions_on_fields`
+ * - ConstGlobalCache:
+ *   - `normal_dot_numerical_flux`
+ * - DataBox:
+ *   - `Tags::Element<volume_dim>`
+ *   - `temporal_id`
+ *   - `Tags::BoundaryDirectionsInterior<volume_dim>`
+ *   - `Tags::BoundaryDirectionsExterior<volume_dim>`
+ *   - `interior<variables_tag>`
+ *   - `exterior<variables_tag>`
+ *   - `interior<normal_dot_numerical_flux::type::argument_tags>`
+ *   - `exterior<normal_dot_numerical_flux::type::argument_tags>`
+ *
+ * DataBox changes:
+ * - Modifies:
+ *   - `exterior<variables_tag>`
+ *   - `Tags::VariablesBoundaryData`
+ */
+template <typename Metavariables>
+struct ImposeHomogeneousDirichletBoundaryConditions {
+  using const_global_cache_tags =
+      tmpl::list<typename Metavariables::normal_dot_numerical_flux>;
+
+  template <typename DbTags, typename... InboxTags, typename ArrayIndex,
+            typename ActionList, typename ParallelComponent>
+  static auto apply(db::DataBox<DbTags>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    using system = typename Metavariables::system;
+    using dirichlet_tags =
+        typename system::impose_boundary_conditions_on_fields;
+    constexpr size_t volume_dim = system::volume_dim;
+
+    // Set the data on exterior (ghost) faces to impose the boundary conditions
+    db::mutate_apply<
```

Comment:
We've generally been using `db::mutate` for inline lambdas and `db::mutate_apply` for externally defined function structs.  I will admit that the two functions are kind of redundant.

Reply:
I copied this from `NumericalAlgorithms/DiscontinuousGalerkin/Actions/ImposeBoundaryConditions.hpp`. Want me to change it?